core/src/main/java/io/paradoxical/cassieq/dataAccess/QueueRepositoryImpl.java

```diff
@@ -29,57 +39,162 @@ public QueueRepositoryImpl(@NonNull final Session session) {
         this.session = session;
     }
 
+
     @Override
-    public void createQueue(@NonNull final QueueDefinition definition) {
+    public void markForDeletion(final QueueDefinition definition) {
+        if(trySetQueueDefinitionStatus(definition.getId(), QueueStatus.Deleting)) {
+            if (trySetQueueNameAvailableStatus(definition, QueueStatus.Inactive)) {
+                logger.with("definition", definition).success("Marked for deletion");
+            }
+        }
+    }
 
-        initializeMonotonicValue(definition.getQueueName());
+    private boolean trySetQueueNameAvailableStatus(final QueueDefinition definition, final QueueStatus status) {
+        final Statement update =
```

Comment:
I was lazy and overloaded queue status here. That's a mistake and I should create a new enum for queue name stairs (available vs not available)